blerry/blerry_model_GVH5184.be

```diff
@@ -0,0 +1,99 @@
+# The 5184 uses a two line method. Probe1/2 are published in one packet.  Probe3/4 in another.  Not always sequential
+def handle_GVH5184(value, trigger, msg)
+  if trigger == details_trigger
+    var this_device = device_config[value['mac']]
+    var p = bytes(value['p'])
+    var i = 0
+    var adv_len = 0
+    var adv_data = bytes('')
+    var adv_type = 0
+    while i < size(p)
+      adv_len = p.get(i,1)
+      adv_type = p.get(i+1,1)
+      adv_data = p[i+2..i+adv_len]
+      if (adv_type == 0xFF) && (adv_len == 0x14)
+        var this_part_data = [adv_data.get(9, -1), adv_data.get(10, -2), adv_data.get(12, -2), adv_data.get(14, -1), adv_data.get(15, -2), adv_data.get(17, -2)]
+        var last_full_data = this_device['last_p']
+        var this_full_data = [ 0, [-1, 0, 0, 0, 0, 0], [-1, 0, 0, 0, 0, 0], 0 ]
+        var seq_num = adv_data.get(8, -1)
+        if last_full_data == bytes('')
+          last_full_data  = [ -1, [-1, 0, 0, 0, 0, 0], [-1, 0, 0, 0, 0, 0], 0 ]
+        else
+          this_full_data = last_full_data
+        end
+        if (this_part_data == last_full_data[seq_num]) && (last_full_data[3] < 31)
+          this_full_data[3]=this_full_data[3]+1
+          device_config[value['mac']]['last_p'] = this_full_data.copy()
+          return 0
+        end
+        this_full_data[0]= adv_data.get(7, -1)
+        this_full_data[3] = 0
+        this_full_data[seq_num]=this_part_data.copy()
+        device_config[value['mac']]['last_p'] = this_full_data.copy()
+        if (this_full_data[1][0] < 0) || (this_full_data[2][0]) < 0
+          return 0
+        end
+        if this_device['discovery'] && !this_device['done_disc']
+          for j: 1 .. 4
+            publish_binary_sensor_discovery(value['mac'], ('Temperature_'+str(j)+'_Status'), 'plug')
+            publish_binary_sensor_discovery(value['mac'], 'Temperature_'+str(j)+'_Alarm', 'heat')
+            publish_sensor_discovery(value['mac'], 'Temperature_'+str(j), 'temperature', '°C')
+            publish_sensor_discovery(value['mac'], 'Temperature_'+str(j)+'_Target', 'temperature', '°C')
+          end
+          publish_sensor_discovery(value['mac'], 'Battery', 'battery', '%')
+          publish_sensor_discovery(value['mac'], 'RSSI', 'signal_strength', 'dB')
+          device_config[value['mac']]['done_disc'] = true
+        end   
+        var output_map = {}
+        output_map['Time'] = tasmota.time_str(tasmota.rtc()['local'])
+        output_map['alias'] = this_device['alias']
+        output_map['mac'] = value['mac']
+        output_map['via_device'] = device_topic
+        output_map['RSSI'] = value['RSSI']
+        if this_device['via_pubs']
+          output_map['Time_via_' + device_topic] = output_map['Time']
+          output_map['RSSI_via_' + device_topic] = output_map['RSSI']
+        end
+        output_map['Battery'] = math.ceil(this_full_data[0]/255.0*100.0)
+        for j:1 .. 2
+          for k:j-1 .. j
```

Comment:
This loop structure kind of confuses me.
I probably would have done like a standard looking loop and done the index calculations based on that
```
for j:1..2
  for k:1..2
    2*(j-1)+k would be the probe's number
  end
end
```